domains/zeto/internal/zeto/handler_events.go

```diff
@@ -149,6 +152,37 @@ func (z *Zeto) handleLockedEvent(ctx context.Context, smtTree *merkleTreeSpec, s
 	return nil
 }
 
+func (z *Zeto) handleIdentityRegisteredEvent(ctx context.Context, smtKycTree *merkleTreeSpec, ev *prototk.OnChainEvent, tokenName string, res *prototk.HandleEventBatchResponse) error {
+	var registered IdentityRegisteredEvent
+	if err := json.Unmarshal([]byte(ev.DataJson), &registered); err == nil {
+		txData, err := decodeTransactionData(ctx, registered.Data)
+		if err != nil || txData == nil {
+			newTxID := uuid.New()
+			txId := pldtypes.Bytes32UUIDFirst16(newTxID)
```

Comment:
Feels like we should sort this out, if not in this PR then shortly after.

I am worried about it being really confusing having these records with a txid, that isn't actually a transaction anywhere in the system.

However, I can see why you've done it - the `state_confirm_records` DB uses `UUID` as the type so you can't put the public TX in there.

I personally think a zero TXID would be better than generating one - which probably means relaxing a restriction in the Paladin code that receives these. But otherwise I do think it would work.

Do you mind taking a look to see if that's a quick change?